### PR TITLE
Ticket5783 parking sequence

### PR DESCRIPTION
**Pull request by John-Holt-Tessella**
https://github.com/ISISComputingGroup/IBEX/issues/5783

### CODE REVIEW

File: ReflectometryServer/test_modules/utils.py [hunk 1]
```diff
@@ -113,3 +114,20 @@ def create_parameter_with_initial_value(init_value, param_class, *args, **kwargs
     setup_autosave({name: init_value}, {})
     kwargs["autosave"] = True
     return param_class(*args, **kwargs)
+
+
+def no_autosave(func):
```
Comment (ThomasLohnert) [inline ReflectometryServer/test_modules/utils.py:119]: Awesome!

**Review by ThomasLohnert — CHANGES_REQUESTED** [review]
Looks good!

There seems to be an issue with move synchronization when you move two axes in a sequence. The axes will move synchronously to the first step in the sequence, but not for the rest. I don't know how much we care but I think we should define the behaviour, ie. either synchronise all the steps or don't synchronise it at all. At the end of the whole sequence the server also throws the error that says the velocity for this axis has previously been restored, let's get rid of that

File: ReflectometryServer/axis.py [hunk 2]
```diff
@@ -49,15 +51,25 @@ class SetRelativeToBeamUpdate:
     relative_to_beam: float
 
 
+@dataclass()
+class ParkingSequenceUpdate:
+    """
+    Event when parking sequence has been updated
+    """
+    parking_sequence: Optional[int]  # the parking sequence we are at the end of
```
[the remark this answers is not in the document]
Comment (John-Holt-Tessella) [reply ReflectometryServer/axis.py:59]: see https://github.com/ISISComputingGroup/EPICS-refl/pull/23/files/524eacd958239a44eb08d2c1df7041bbb075fe84#diff-c24982c71051485fde2eca3c6778b9ae3d9fbd10c3e99d8d0503da8e054477faR290

```
For RBV set the index of the parking sequence index just reached by the motor; These must be generated in order
For SP set the index of the parking sequence the motor should be going to
```

File: ReflectometryServer/exceptions.py [hunk 3]
```diff
@@ -23,3 +23,23 @@ def __init__(self, err):
 
     def __str__(self):
         return self.message
+
+
+class BeamlineConfigurationParkAutosaveInvalidException(Exception):
```
[the remark this answers is not in the document]
Comment (John-Holt-Tessella) [reply ReflectometryServer/exceptions.py:28]: Add documentation to https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reflectometry-Troubleshooting-and-FAQ

**Comment by John-Holt-Tessella** [comment]
Turns out ticket to make velocity cache warning better had different implementation so I have used that instead of removing it as we spoke about (sorry)

**Review by ThomasLohnert — CHANGES_REQUESTED** [review]
Found some odd behaviour using a config that has a component with the following defined parking behaviour:
```
    add_driver(IocDriver(test_component, ChangeAxis.HEIGHT, MotorPVWrapper("MOT:MTR0107"),
                         out_of_beam_positions=[OutOfBeamSequence([2, -4])]))
    add_driver(IocDriver(test_component, ChangeAxis.PHI, MotorPVWrapper("MOT:MTR0108"),
                         out_of_beam_positions=[OutOfBeamSequence([3, -6])]))
```

1. Move component out of beam
2. Restart IOC
3. Set inbeam param to `IN` and move parameter individually --> Axes do not move (would expect them to go to intermediate park position, then to their offset SPs)
4. Set inbeam param to `OUT` --> Axes go to intermediate step, then back to their initial out position (would expect them not to move, but this is less critical)
5. Set inbeam param to `IN` and move all --> parking axes go straight to axis offset SPs (would expect them to go to intermediate park position, then to their offset SPs)

**Review by ThomasLohnert — APPROVED** [review]

**Pull request merged**